test/database/service/BookingServiceTest.java

```diff
@@ -0,0 +1,42 @@
+package database.service;
+
+import static org.junit.Assert.assertEquals;
+import java.math.BigDecimal;
+import java.sql.SQLException;
+import java.time.LocalDateTime;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import database.dao.BookingDataDAO;
+import database.dto.BookingDataDTO;
+import database.dto.PaymentMethodDTO;
+import service.BookingService;
+
+@RunWith(JUnit4.class)
+public class BookingServiceTest {
+	
+	@Mock
+	private BookingDataDAO bookingDataDAO;
+	
+	private BookingService bookingService;
+	
+	@Before 
+	public void setUp() throws SQLException {
+		MockitoAnnotations.openMocks(this);
+		bookingService = new BookingService(bookingDataDAO);
+	}
+	
+	@Test
+	public void testSaveBooking() {
+		BookingDataDTO bookingDataDTO=new BookingDataDTO(LocalDateTime.now(),LocalDateTime.of(2023,12,12,10,30),PaymentMethodDTO.CASH,new BigDecimal("6000"));
+		bookingDataDTO.setId(123);
+		Mockito.doReturn(bookingDataDTO).when(bookingDataDAO).save(bookingDataDTO);
```

Comment:
This test isn't working because you're creating a new instance of `BookingDataDTO` inside of `BookingDataDAO.save`, which doesn't match the `.when(bookingDataDTO)`.

Please research about the Mockito.any clause.

![image](https://github.com/yasalinasbel/challenge-one-alura-hotel-latam/assets/136126143/457a2f49-7a79-474c-a2b3-345b27df1628)